src/client.rs

```diff
@@ -313,8 +313,18 @@ impl RequestBuilder {
                         headers.set(Referer(url.to_string()));
                         urls.push(url);
                         let action = check_redirect(&client.redirect_policy.lock().unwrap(), &loc, &urls);
+
                         match action {
-                            redirect::Action::Follow => loc,
+                            redirect::Action::Follow => {
+                                let cross_host = loc.host().unwrap() != urls.last().unwrap().host().unwrap();
+                                if cross_host {
+                                    headers.remove::<Authorization<::header::Basic>>();
+                                    headers.remove::<Authorization<String>>();
+                                    headers.remove::<Authorization<::header::Bearer>>();
```

Comment:
These 3 lines actually all do the same thing (they all just remove from the internal header map the entire entry for `Authorization`).